test/unit/org/apache/cassandra/cql3/validation/operations/AlterTest.java

```diff
@@ -269,6 +276,59 @@ public void testCreateAlterKeyspaces() throws Throwable
                                   "max_threshold", "32")));
     }
 
+    @Test
+    public void testCreateAlterKeyspacesRFWarnings() throws Throwable
+    {
+        requireNetwork();
+        Thread.sleep(1000); // Hack: requireNetwork() will clear warnings async so give some time to complete
+
+        // NTS
+        ClientWarn.instance.captureWarnings();
+        String ks = createKeyspace("CREATE KEYSPACE %s WITH replication = {'class' : 'NetworkTopologyStrategy', '" + DATA_CENTER + "' : 3 }");
+        List<String> warnings = ClientWarn.instance.getWarnings();
+        assertEquals(1, warnings.size());
+        Assertions.assertThat(warnings.get(0)).contains("Your replication factor 3 for keyspace " + ks + " is higher than the number of nodes 1 for datacenter " + DATA_CENTER);
+
+        ClientWarn.instance.captureWarnings();
+        execute("CREATE TABLE " + ks + ".t (k int PRIMARY KEY, v int)");
+        warnings = ClientWarn.instance.getWarnings();
+        assertNull(warnings);
+
+        ClientWarn.instance.captureWarnings();
```

Comment:
with so many cases maybe we can extract some methods 
```
private void expectCreateKeyspaceWarning(ddl, warning)
{
    List<String> warnings = ClientWarn.instance.getWarnings();
   createKeyspace(ddl);
    assertEquals(1, warnings.size());
    Assertions.assertThat(warnings.get(0)).contains(warning);    
}
```
and
```
private void expectNoWarning(...)
```
so that test would become a list of checks?
```
expectWarning(...)
expectNoWarning(....)
```

Reply:
+1 if it were a test class wide util, but being a small and single method... meh I see it as a nit imo.

Reply:
up to you